### ISSUE TITLE
API for pre-read flight data

### ISSUE THREAD

**Issue by tjlang** [nguy/AWOT#18]
Sometimes all we have from an aircraft is lat, lon, alt, and time. Or the data are otherwise too customized to bother with a universal reader. Maybe add a method that accepts a dictionary of various variables that someone else has already read in, and constructs the right AWOT objects for further analysis?

**Comment by nguy** [comment]
That is what the `mapping_dict` keyword does. It accepts a dictionary instead of either using a default chosen when platform is specified or by doing a straight copy of variable names.

For example you could:

``` python
name_map = {'time': 'Time', 'latitude': 'LAT3.gps', 'longitude': 'longavgrx', 'altitude': 'ztrue'}
f=awot.io.read_netcdf(filename, mapping_dict=name_map)
```

and the returne AWOT object uses the names and maps them to the variables.

**Comment by nguy** [comment]
Obviously one downfall currently is the documentation.... :(

**Comment by tjlang** [comment]
Cool, I'll check that out. Although what I am thinking of is more like the user reading the data themselves using whatever (pandas, numpy.loadtxt, etc.), then supply a data dictionary to AWOT to create the f object. That way they can access the AWOT tools but don't need to worry about writing a custom reader for AWOT every time they discover a new aircraft data format.

**Comment by nguy** [comment]
Oh, I see. I think with a little retooling, this could be done. My first thought would be to provide a `to_awot` function that let's the user easily make an awot object for plotting, calculations, etc. Probably in the `awot.io` space?

I'm also thinking now of moving all of the name maps to a separate _common_ file that can be called from the netCDF reader (including the T-28 you produced).

**Comment by tjlang** [comment]
These all sound like good options.

**Comment by nguy** [comment]
For a while I considered switching to Pandas as an interface reader, but with a lot of 2D arrays, this becomes a little trickier. I use pandas a lot for reading some aircraft data, so this might be a nice way to integrate Pandas in without the extra dependency.

**Comment by nguy** [comment]
This was addressed in PR #20 , let me know if you think this is sufficient for the time being or whether increased functionality is needed.

**Comment by nguy** [comment]
Closing the issue since no traffic and it _should_ be fixed.
